services/requester/requester.go

```diff
@@ -240,7 +240,7 @@ func (e *EVM) signAndSend(ctx context.Context, script []byte, args ...cadence.Va
 
 			e.logger.Debug().
 				Str("flow-id", id.String()).
-				Str("events", fmt.Sprintf("%v", res.Events)).
```

Comment:
The `Str` version produces the following:
```json
{
    "level": "debug",
    "component": "requester",
    "flow-id": "3e49dadc7d4b0c330fc04ea3fd060d13e87954ef5a14330fc6982908343aec27",
    "events": "[A.0ae53cb6e3f42a79.FlowToken.TokensWithdrawn: 0x07c4914432d2c81bf0c15ad6e0cf4c6a1a45785e6008050db25436ba11728747 A.ee82856bf20e2aa6.FungibleToken.Withdrawn: 0x209cf67ec24826c7eead52e1832926a828b60180a7835bde3e67a192352f88ed evm.TransactionExecuted: 0xdb948ce6d504beb918c03dc37c150999e3ca5e9a58c73eaec42d71f0d86221b6 evm.BlockExecuted: 0x4deec392375ab3def6e513f986e9e3978b964ed448c7896aebdce5a6ea04bef0 A.f8d6e0586b0a20c7.EVM.CadenceOwnedAccountCreated: 0xe39dbfb45803c362e118cd259469f9982011e70f2e1116888823afe083b5be6f evm.TransactionExecuted: 0x16e0672f04cf8278e691c8cda593156e39b63976d89d4775a1aa9beadd8305dd evm.BlockExecuted: 0x44c3302d500289a198690761d8c0fcc91b3e29b9e0951d2f964a0345a4a3175d]",
    "time": "2024-03-11T14:51:30+02:00",
    "message": "flow transaction executed successfully"
}
```
So its basically just the event name and event ID.

The `Interface` version produces:
```json
{
    "level": "debug",
    "component": "requester",
    "flow-id": "e0c986f1a53ee36cff6882e06116894e3e2bd1ebe55b64f8067f67093bcce9b6",
    "events": [
        {
            "Type": "evm.TransactionExecuted",
            "TransactionID": [
                224,
                201,
                134,
                241,
                165,
                62,
                227,
                108,
                255,
                104,
                130,
                224,
                97,
                22,
                137,
                78,
                62,
                43,
                209,
                235,
                229,
                91,
                100,
                248,
                6,
                127,
                103,
                9,
                59,
                204,
                233,
                182
            ],
            "TransactionIndex": 0,
            "EventIndex": 0,
            "Value": {
                "EventType": {
                    "Location": {
                        "Type": "EVMLocation"
                    },
                    "QualifiedIdentifier": "TransactionExecuted",
                    "Fields": [
                        {
                            "Type": 48,
                            "Identifier": "blockHeight"
                        },
                        {
                            "Type": 8,
                            "Identifier": "blockHash"
                        },
                        {
                            "Type": 8,
                            "Identifier": "transactionHash"
                        },
                        {
                            "Type": 8,
                            "Identifier": "transaction"
                        },
                        {
                            "Type": 6,
                            "Identifier": "failed"
                        },
                        {
                            "Type": 8,
                            "Identifier": "vmError"
                        },
                        {
                            "Type": 45,
                            "Identifier": "transactionType"
                        },
                        {
                            "Type": 48,
                            "Identifier": "gasConsumed"
                        },
                        {
                            "Type": 8,
                            "Identifier": "deployedContractAddress"
                        },
                        {
                            "Type": 8,
                            "Identifier": "returnedValue"
                        },
                        {
                            "Type": 8,
                            "Identifier": "logs"
                        }
                    ],
                    "Initializer": null
                },
                "Fields": [
                    3,
                    "0x35b5a97d6e7b0f7b220b8eb922b7067c484faf633c89e612882285d47b0390a8",
                    "0xe414f90fea2aebd75e8c0b3b6a4a0c9928e86c16ea724343d884f40bfe2c4c6b",
                    "f9015880808301e8488080b901086060604052341561000f57600080fd5b60eb8061001d6000396000f300606060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063c6888fa1146044575b600080fd5b3415604e57600080fd5b606260048080359060200190919050506078565b6040518082815260200191505060405180910390f35b60007f24abdb5865df5079dcc5ac590ff6f01d5c16edbc5fab4e195d9febd1114503da600783026040518082815260200191505060405180910390a16007820290509190505600a165627a7a7230582040383f19d9f65246752244189b02f56e8d0980ed44e7a56c0b200458caad20bb002982052fa09c05a7389284dc02b356ec7dee8a023c5efd3a9d844fa3c481882684b0640866a057e96d0a71a857ed509bb2b7333e78b2408574b8cc7f51238f25c58812662653",
                    false,
                    "",
                    0,
                    103856,
                    "99466ed2e37b892a2ee3e9cd55a98b68f5735db2",
                    "606060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063c6888fa1146044575b600080fd5b3415604e57600080fd5b606260048080359060200190919050506078565b6040518082815260200191505060405180910390f35b60007f24abdb5865df5079dcc5ac590ff6f01d5c16edbc5fab4e195d9febd1114503da600783026040518082815260200191505060405180910390a16007820290509190505600a165627a7a7230582040383f19d9f65246752244189b02f56e8d0980ed44e7a56c0b200458caad20bb0029",
                    ""
                ]
            },
            "Payload": "2IGCgdiig0B3ZXZtLlRyYW5zYWN0aW9uRXhlY3V0ZWSLgmtibG9ja0hlaWdodNiJD4JpYmxvY2tIYXNo2IkBgm90cmFuc2FjdGlvbkhhc2jYiQGCa3RyYW5zYWN0aW9u2IkBgmZmYWlsZWTYiQCCZ3ZtRXJyb3LYiQGCb3RyYW5zYWN0aW9uVHlwZdiJDIJrZ2FzQ29uc3VtZWTYiQ+Cd2RlcGxveWVkQ29udHJhY3RBZGRyZXNz2IkBgm1yZXR1cm5lZFZhbHVl2IkBgmRsb2dz2IkBgtiIQIsDeEIweDM1YjVhOTdkNmU3YjBmN2IyMjBiOGViOTIyYjcwNjdjNDg0ZmFmNjMzYzg5ZTYxMjg4MjI4NWQ0N2IwMzkwYTh4QjB4ZTQxNGY5MGZlYTJhZWJkNzVlOGMwYjNiNmE0YTBjOTkyOGU4NmMxNmVhNzI0MzQzZDg4NGY0MGJmZTJjNGM2YnkCtmY5MDE1ODgwODA4MzAxZTg0ODgwODBiOTAxMDg2MDYwNjA0MDUyMzQxNTYxMDAwZjU3NjAwMDgwZmQ1YjYwZWI4MDYxMDAxZDYwMDAzOTYwMDBmMzAwNjA2MDYwNDA1MjYwMDQzNjEwNjAzZjU3NjAwMDM1N2MwMTAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwOTAwNDYzZmZmZmZmZmYxNjgwNjNjNjg4OGZhMTE0NjA0NDU3NWI2MDAwODBmZDViMzQxNTYwNGU1NzYwMDA4MGZkNWI2MDYyNjAwNDgwODAzNTkwNjAyMDAxOTA5MTkwNTA1MDYwNzg1NjViNjA0MDUxODA4MjgxNTI2MDIwMDE5MTUwNTA2MDQwNTE4MDkxMDM5MGYzNWI2MDAwN2YyNGFiZGI1ODY1ZGY1MDc5ZGNjNWFjNTkwZmY2ZjAxZDVjMTZlZGJjNWZhYjRlMTk1ZDlmZWJkMTExNDUwM2RhNjAwNzgzMDI2MDQwNTE4MDgyODE1MjYwMjAwMTkxNTA1MDYwNDA1MTgwOTEwMzkwYTE2MDA3ODIwMjkwNTA5MTkwNTA1NjAwYTE2NTYyN2E3YTcyMzA1ODIwNDAzODNmMTlkOWY2NTI0Njc1MjI0NDE4OWIwMmY1NmU4ZDA5ODBlZDQ0ZTdhNTZjMGIyMDA0NThjYWFkMjBiYjAwMjk4MjA1MmZhMDljMDVhNzM4OTI4NGRjMDJiMzU2ZWM3ZGVlOGEwMjNjNWVmZDNhOWQ4NDRmYTNjNDgxODgyNjg0YjA2NDA4NjZhMDU3ZTk2ZDBhNzFhODU3ZWQ1MDliYjJiNzMzM2U3OGIyNDA4NTc0YjhjYzdmNTEyMzhmMjVjNTg4MTI2NjI2NTP0YAAaAAGVsHgoOTk0NjZlZDJlMzdiODkyYTJlZTNlOWNkNTVhOThiNjhmNTczNWRiMnkB1jYwNjA2MDQwNTI2MDA0MzYxMDYwM2Y1NzYwMDAzNTdjMDEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDkwMDQ2M2ZmZmZmZmZmMTY4MDYzYzY4ODhmYTExNDYwNDQ1NzViNjAwMDgwZmQ1YjM0MTU2MDRlNTc2MDAwODBmZDViNjA2MjYwMDQ4MDgwMzU5MDYwMjAwMTkwOTE5MDUwNTA2MDc4NTY1YjYwNDA1MTgwODI4MTUyNjAyMDAxOTE1MDUwNjA0MDUxODA5MTAzOTBmMzViNjAwMDdmMjRhYmRiNTg2NWRmNTA3OWRjYzVhYzU5MGZmNmYwMWQ1YzE2ZWRiYzVmYWI0ZTE5NWQ5ZmViZDExMTQ1MDNkYTYwMDc4MzAyNjA0MDUxODA4MjgxNTI2MDIwMDE5MTUwNTA2MDQwNTE4MDkxMDM5MGExNjAwNzgyMDI5MDUwOTE5MDUwNTYwMGExNjU2MjdhN2E3MjMwNTgyMDQwMzgzZjE5ZDlmNjUyNDY3NTIyNDQxODliMDJmNTZlOGQwOTgwZWQ0NGU3YTU2YzBiMjAwNDU4Y2FhZDIwYmIwMDI5YA=="
        },
        {
            "Type": "evm.BlockExecuted",
            "TransactionID": [
                224,
                201,
                134,
                241,
                165,
                62,
                227,
                108,
                255,
                104,
                130,
                224,
                97,
                22,
                137,
                78,
                62,
                43,
                209,
                235,
                229,
                91,
                100,
                248,
                6,
                127,
                103,
                9,
                59,
                204,
                233,
                182
            ],
            "TransactionIndex": 0,
            "EventIndex": 1,
            "Value": {
                "EventType": {
                    "Location": {
                        "Type": "EVMLocation"
                    },
                    "QualifiedIdentifier": "BlockExecuted",
                    "Fields": [
                        {
                            "Type": 48,
                            "Identifier": "height"
                        },
                        {
                            "Type": 8,
                            "Identifier": "hash"
                        },
                        {
                            "Type": 36,
                            "Identifier": "totalSupply"
                        },
                        {
                            "Type": 8,
                            "Identifier": "parentHash"
                        },
                        {
                            "Type": 8,
                            "Identifier": "receiptRoot"
                        },
                        {
                            "Type": {
                                "ElementType": 8
                            },
                            "Identifier": "transactionHashes"
                        }
                    ],
                    "Initializer": null
                },
                "Fields": [
                    3,
                    "0x35b5a97d6e7b0f7b220b8eb922b7067c484faf633c89e612882285d47b0390a8",
                    {
                        "Value": 10000000000
                    },
                    "0xae483432417abf0d8ac71e850caa868434467b64cd754260375bf013b8389551",
                    "0x0000000000000000000000000000000000000000000000000000000000000000",
                    {
                        "ArrayType": {
                            "ElementType": 8
                        },
                        "Values": [
                            "0xe414f90fea2aebd75e8c0b3b6a4a0c9928e86c16ea724343d884f40bfe2c4c6b"
                        ]
                    }
                ]
            },
            "Payload": "2IGCgdiig0BxZXZtLkJsb2NrRXhlY3V0ZWSGgmZoZWlnaHTYiQ+CZGhhc2jYiQGCa3RvdGFsU3VwcGx52IkEgmpwYXJlbnRIYXNo2IkBgmtyZWNlaXB0Um9vdNiJAYJxdHJhbnNhY3Rpb25IYXNoZXPYi9iJAYLYiECGA3hCMHgzNWI1YTk3ZDZlN2IwZjdiMjIwYjhlYjkyMmI3MDY3YzQ4NGZhZjYzM2M4OWU2MTI4ODIyODVkNDdiMDM5MGE4wkUCVAvkAHhCMHhhZTQ4MzQzMjQxN2FiZjBkOGFjNzFlODUwY2FhODY4NDM0NDY3YjY0Y2Q3NTQyNjAzNzViZjAxM2I4Mzg5NTUxeEIweDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDCBeEIweGU0MTRmOTBmZWEyYWViZDc1ZThjMGIzYjZhNGEwYzk5MjhlODZjMTZlYTcyNDM0M2Q4ODRmNDBiZmUyYzRjNmI="
        }
    ],
    "time": "2024-03-11T14:42:54+02:00",
    "message": "flow transaction executed successfully"
}
```

Which is basically every single field from the event payloads. Which of the two we prefer? The second can produce quite a lengthy output.